### PR TITLE
Bring Spring Boot Guide home

### PR DESCRIPTION
**Pull request by ge0ffrey**
Refactored the Quick Start section of the OptaPlanner docs into 3 sections:

- Quarkus (now just a link to quarkus.io, content coming soon)
- Spring Boot (that guide that was hidden in that PR we send to the spring guys and they rejected)
- Java (the cloud balancing guide which needs some love)

Motivation to put Quarkus and Spring Boot before Java: realistically, most serious users tends to require REST etc.

### CODE REVIEW

**Comment by rsynek** [comment]
@ge0ffrey have you done any significant changes to the content of the Spring Boot Guide since you targetted it for the spring.io? I checked the Initializr's output, the text, links, etc., but unless there were changes in the content itself, checking the instructions step by step is probably not necessary (already done that before).

File: optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc [hunk 1]
```diff
@@ -0,0 +1,1057 @@
...
+        assertTrue(timeTable.getScore().isFeasible());
+    }
+
+}
+----
+====
+
+. Build an attractive web UI on top of these REST methods to visualize the timetable.
+
+Take a look at the example's source code to see how this all turns out.
```
Comment (rsynek) [inline optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc:1057]: I haven't noticed any link for the example's source code. On the other hand, if this documentation ends up in a product doc, we should not refer to any upstream resources there.

**Comment by ge0ffrey** [comment]
> @ge0ffrey have you done any significant changes to the content of the Spring Boot Guide since you targetted it for the spring.io? I checked the Initializr's output, the text, links, etc., but unless there were changes in the content itself, checking the instructions step by step is probably not necessary (already done that before).

No, there weren't any significant changes, besides embedding the java files to make it easier to productize.

File: optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc (hunk 1 above)
Comment (ge0ffrey) [reply optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc:1057]: Yea, but we could refer to the download zip. I 'll adjust that in the next PR.

**Pull request merged**